ovirt_imageio/tool/__init__.py

```diff
@@ -0,0 +1,101 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Tool for transferring disk images.
+"""
+
+import argparse
+import getpass
+
+from contextlib import closing
+
+from .. import client
+from . import ovirt
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Transfer disk images")
+    parser.set_defaults(
+        command=lambda _: parser.print_help())
+
+    commands = parser.add_subparsers(title="commands")
+
+    cmd = commands.add_parser("download-disk", help="Download disk")
+    cmd.set_defaults(command=download_disk)
+
+    add_common_options(cmd)
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
+        help="Download image format (default qcow2).")
+
+    cmd.add_argument(
+        "disk_id",
+        help="Disk ID to download.")
+
+    cmd.add_argument(
+        "filename",
+        help="Target filename.")
+
+    args = parser.parse_args()
+
+    # XXX Configure logging.
+    # XXX Setup signals.
+
+    args.command(args)
+
+
+def add_common_options(cmd):
+    cmd.add_argument(
+        "--engine-url",
+        help="ovirt-engine URL.")
+
+    cmd.add_argument(
+        "--username",
+        help="ovirt-engine username.")
+
+    cmd.add_argument(
+        "--password-file",
+        help="Read ovirt-engine password from file.")
+
+    cmd.add_argument(
+        "--cafile",
+        help="Path to ovirt-engine CA certificate")
+
+
+def fill_password(args):
+    if args.password_file:
+        with open(args.password_file) as f:
+            args.password = f.read().rstrip("\n")
```

Comment:
Is that common practice in python, to [over]write cli-arguments like that? I think in some languages that is discouraged.

Reply:
args.password is not really an argument that a user can set. This is just an attribute in a Namespace object. But next version does not override it since we read the password just before it is needed.